packages/server/lib/cypress.ts

```diff
@@ -9,14 +9,19 @@ require('./environment')
 // essentially do it all again when we boot the correct
 // mode.
 
-const Promise = require('bluebird')
-const debug = require('debug')('cypress:server:cypress')
-const { getPublicConfigKeys } = require('@packages/config')
-const argsUtils = require('./util/args')
-const { telemetry } = require('@packages/telemetry')
-const { getCtx, hasCtx } = require('@packages/data-context')
+import Promise from 'bluebird'
+import Debug from 'debug'
+import { getPublicConfigKeys } from '@packages/config'
+import argsUtils from './util/args'
+import { telemetry } from '@packages/telemetry'
+import { getCtx, hasCtx } from '@packages/data-context'
+import type { AllCypressErrorObj } from '@packages/errors/src/errors'
 
-const warning = (code, args) => {
+const debug = Debug('cypress:server:cypress')
+
+type Mode = 'exit' | 'info' | 'interactive' | 'pkg' | 'record' | 'results' | 'run' | 'smokeTest' | 'version' | 'returnPkg' | 'exitWithCode'
```

Comment:
This type may have more types than are supported. It was hard to tell the definitive list by how this is called.